File: 22/Day13.py
import re
Advent = open("C:\\Users\\alexf\\OneDrive\\Documents\\AdventofCode\\22\\Day13.txt", "r")
Advent = Advent.readlines()
Array = []
def GetCompare(text):
    AllVals = []
    index = 0
    while True:
        str = ""
        arr = 0
        for c in text[index:-1]:
            index += 1
            if c == "[":
                arr += 1
            if c == "]":
                arr -= 1
            if c == "," and arr == 0:
                AllVals.append(str)
                break
            str += c
        if index == (len(text) - 1):
            break
    AllVals.append(str)
    #print(AllVals)
    return AllVals

turn = 0
for line in Advent:
    if not line.isspace():
        arrayiter = line.__iter__()
        line = line.replace('\n','')
        #line = line.removeprefix("[").removesuffix("]")
        #line = [line]
        #iter = line.__iter__()
        #print(line.__len__())
        if line.__contains__(","):
            if turn == 0:
                left = GetCompare(line[1:])
                turn = 1
            else:
                right = GetCompare(line[1:])
                turn = 0
        else:
            if turn == 0:
                left = [line]
                turn = 1
            else:
                right = [line]
                turn = 0
    else:
        ind = 0
        total = 0
        if left < right:
            print(f"{left}\n{right}\nsuccess\n")
            continue
        elif right < left:
            print(f"{left}\n{right}\nfailure\n")
            continue
        # could probably do this with just left count
        if left.__len__() > right.__len__():
            total = left.__len__()
        else:
            total = right.__len__()
        print(left)
        print(right)
        while ind < total:
            #print(ind)
            newleft = left[ind]
            newright = right[ind]
            while newleft.__contains__(",") or newright.__contains__(","):
                if newleft.__contains__(","):
                    newleft = GetCompare(newleft)
                    if newright.__contains__(","):
                        newright = GetCompare(newright)
                    else:
                        newright = [newright]
                else:
                    newleft = [newleft]
                    newright = GetCompare(newright)
            if newleft < newright:
                print("success")
                break
            elif newleft > newright:
                print("failure")
                break
            ind = ind + 1
            if (ind + 1) == left.__len__():
                print("success")
                break
            elif (ind + 1) > right.__len__():
                print("failure")
                break
        
    
    """
    if not line.isspace():
        if line.__contains__(","):
            if turn == 0:
                left = GetCompare(line[1:-1])
                turn += 1
            else:
                right = GetCompare(line[1:-1])
                turn = 0
        else:
            if turn == 0:
                left = line[0:-1]
                turn += 1
            else:
                right = line[0:-1]
                turn = 0
    else:
        valcount = 0
        #print(f'This is pair: {valcount + 1}')
        lnumber = re.sub('[\D]', '', left[valcount])
        rnumber = re.sub('[\D]', '', right[valcount])
        #print(left)
        #print(right)
        #if not (left[valcount].__contains__(",")) and not (right[valcount].__contains__(",")) :
            #if lnumber.__len__() != 0:
                #print(lnumber[0])
           # else:
                #print(left)
            #if rnumber.__len__() != 0:
                #print(rnumber[0])
            #else:
                #print(right)
        #while (valcount < left.count(int)):
            
            ####valcount += 1
  """          

#print(line)
#print(line.replace("[","").replace("]",""))

File: 22/C#/Day13.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Security.Cryptography.X509Certificates;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;

namespace Advent22
{
    class entry : IComparable<entry>
    {
        public readonly string content;
        public string GetContent() { return content; }
        
        public int CompareTo(entry obj)
        {
            string otherContent = obj.GetContent();
            otherContent = otherContent.Substring(1, otherContent.Length - 2);

            // PUT IN THE SORTING FUNCTION HERE!!
            
            // If empty arrays, the bigger array counts as a higher value
            if (!Regex.Match(otherContent, @"\d").Success && !Regex.Match(this.content, @"\d").Success) {
                if (this.content.Length > otherContent.Length)
                {
                    return 1;
                }
                else
                {
                    return -1;
                }
            }

            // If the above failed, and one of the string does not contain a number, return value
            if (!Regex.Match(otherContent, @"\d").Success)
            {
                return 1;
            }
            if (!Regex.Match(this.content, @"\d").Success)
            {
                return -1;
            }

            List<string> thisList = this.content.Replace("[", "").Replace("]", "").Split(",").ToList();
            List<string> otherList = otherContent.Replace("[", "").Replace("]", "").Split(",").ToList();
            int max;

            if (thisList.Count > otherList.Count) max = thisList.Count; else max = otherList.Count;

            for (int i = 0; i < max; i++)
            {
                if (i == thisList.Count) return -1;
                if (i == otherList.Count) return 1;
                if (thisList[i].Length == 0 && otherList[i].Length != 0) return -1;
                if (thisList[i].Length != 0 && otherList[i].Length == 0) return 1;
                if (thisList[i].Length == 0 && otherList[i].Length == 0) continue;
                if (int.Parse(thisList[i]) == int.Parse(otherList[i])) continue;

                if (int.Parse(thisList[i]) > int.Parse(otherList[i]))
                {
                    return 1;
                }
                else
                {
                    return -1;
                }
            }

            //
            return -1;
        }
        public entry(string content)
        {
            this.content = content;
        }
    }
    internal class Day13
    {
        readonly List<string> File = System.IO.File.ReadLines("C:\\Users\\alexf\\OneDrive\\Documents\\AdventofCode\\AdventofCode\\22\\Day13.txt").ToList();
        readonly List<entry> entries= new ();
        public Day13() 
        { 
            for (int i = 0; i < File.Count; i++)
            {
                if ((i+1) % 3 != 0)
                {
                    entries.Add(new entry(File[i]));
                }
            }
            entries.Add(new entry("[[2]]"));
            entries.Add(new entry("[[6]]"));
            entries.Sort();
            //entries.ForEach(x => Console.WriteLine(x.GetContent()));
            int two = entries.IndexOf(entries.Find(x => x.content == "[[2]]")) + 1;
            int six = entries.IndexOf(entries.Find(x => x.content == "[[6]]")) + 1;
            Console.WriteLine(two * six);
        }
    }
}

// 19278 // didn't index correctly, dick puzzle
//19570?? Yes!!